### PR TITLE
Healthcheck gives correct result even behind proxy

### PR DESCRIPTION
**Pull request by MatthiasJobst**
## Changes
For environments that use proxies, mostly corporate environments, the health check did not work. Proxies are not necessary for the health check so the proxies are disabled.
The health check runs in the docker container and makes a request to a specific address. If that request is routed through a proxy it leaves the docker container. This can cause two types of problems:
- The proxy does not know the ip address and thus the request never terminates
- Outside of the container the ports can change and the request will not reach the proper port.

## Issue ticket

Closed [1962](https://github.com/jupyter/docker-stacks/issues/1962)

## Checklist (especially for first-time contributors)

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests
- [x] I will try not to use force-push to make the review process easier for reviewers
- [-] I have updated the documentation for significant changes

### CODE REVIEW

File: base-notebook/docker_healthcheck.py [hunk 1]
```diff
@@ -16,6 +16,11 @@
 url = json.loads(json_file.read_bytes())["url"]
 url = url + "api"
 
-r = requests.get(url, verify=False)  # request without SSL verification
+proxies = {
+    "http": None,
+    "https": None,
+}
```
Comment (mathbunnyru) [inline base-notebook/docker_healthcheck.py:22]:
```suggestion
proxies = {
    "http": "",
    "https": "",
}
```

This should solve mypy issue.
There seems to be some difference  though.
https://stackoverflow.com/a/72529216/4881441

**Review by mathbunnyru — APPROVED** [review]
Overall, LGTM and thank you for adding a test 👍

**Comment by MatthiasJobst** [comment]
The tests that fail do not have anything to do with the test I added. Interestingly they seem to fail without the changes too? 
Maybe some deprecation issue with notebook?

**Comment by mathbunnyru** [comment]
> The tests that fail do not have anything to do with the test I added. Interestingly they seem to fail without the changes too?
> Maybe some deprecation issue with notebook?

Let's hope it was a spurious fail. I restarted failed GitHub jobs.

**Comment by mathbunnyru** [comment]
I guess the reason is that for some reason `mamba` decides to downgrade `jupyterlab` and `notebook`:
```
#6 394.0   Change:
#6 394.0 ────────────────────────────────────────────────────────────────────────────────────────
#6 394.0 
#6 394.0   - _openmp_mutex                  4.5  2_gnu                    conda-forge      24kB
#6 394.0   + _openmp_mutex                  4.5  2_kmp_llvm               conda-forge       6kB
#6 394.0 
#6 394.0   Downgrade:
#6 394.0 ────────────────────────────────────────────────────────────────────────────────────────
#6 394.0 
#6 394.0   - notebook                     7.0.2  pyhd8ed1ab_0             conda-forge       3MB
#6 394.0   + notebook                     6.5.4  pyha770c72_0             conda-forge     306kB
#6 394.0   - jupyterlab                   4.0.4  pyhd8ed1ab_0             conda-forge       6MB
#6 394.0   + jupyterlab                   3.6.5  pyhd8ed1ab_0             conda-forge       5MB
```

I also see a warning because of `jupyter_lsp` not being able to run.
Probably jupyter app doesn't start properly after downgrade and we get an error.
It's definitely your change and I'll need to fix it in the `main` branch.
I will take a look in a few days in more details.

**Comment by mathbunnyru** [comment]
@MatthiasJobst I fixed (actually, mitigated) the issue in the main branch and merged it here.
Hopefully, tests will pass and then I'll merge this PR.

**Comment by MatthiasJobst** [comment]
Thanks. That was quick and solved the issue.

**Pull request merged**